### ISSUE TITLE
possible to supply answers object at the start of prompt

### ISSUE THREAD

**Issue by southpolesteve** [SBoudrias/Inquirer.js#350]
I've been splunking through the code trying to figure out if this is possible, but no luck so far. I would like to be able to supply inquirer with a starting point. This would allow users to supply configuration via env variables or CLI flags and prompt will use its `when` functionality to decide what else it should ask for.  

Example of what I am trying to accomplish:

``` js
const flags = parseCLIFlags()

inquirier.prompt(flags, questions)
```

By suppling a `answers` object up front, prompt would have a starting point and only ask for things the user has not already supplied.

Apologies in advance if this has been asked before. I couldn't find anything searching through the code or old issues.

**Comment by SBoudrias** [comment]
Why not something of your own:

```js
const flags = parseCLIFlags()

inquirer.prompt(excludePreAnsweredQuestions(questions, flags));
```

Then this doesn't need to live in Inquirer. It's just a function adapter to your needs.

**Comment by southpolesteve** [comment]
That was my initial solution, but as the prompts become more complicated (specifically more logic in the `when` functions) I found myself with weird edge cases. I did tried an attempt at replacing the answers object on the prototype but wasn't able to make it work.

Just spliting up the prompts or try and simplifying the question hierarchy may just be the best solution here. I could also walk and execute the `when` functions but that may still lead to edge cases.
